### PR TITLE
[WIP] Get related docs by field

### PR DESCRIPTION
**Pull request by Libisch**
add `get_document_related_docs_by_fields` function to api.py
add test to tests/test_clearmash_api

expected: pass test
actual: test fails:
```
(ENV) Libis-MacBook-Pro:mojp-dbs-pipelines libi$ py.test tests/test_clearmash_api.py -svv
================================================= test session starts ==================================================
platform darwin -- Python 3.6.1, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /Users/libi/mojp-dbs-pipelines/ENV/bin/python3.6
cachedir: .cache
rootdir: /Users/libi/mojp-dbs-pipelines, inifile: tox.ini
plugins: celery-4.0.2
collecting 0 items/Users/libi/mojp-dbs-pipelines/ENV/lib/python3.6/site-packages/dotenv/main.py:24: UserWarning: Not loading  - it doesn't exist.
  warnings.warn("Not loading %s - it doesn't exist." % dotenv_path)
collected 6 items 

tests/test_clearmash_api.py::test_invalid_call INFO:api:_wcm_api_call(/invalid/path, None)
PASSED
tests/test_clearmash_api.py::test_get_root_folder INFO:api:_wcm_api_call(/CommunityPage/Folder/Root, None)
PASSED
tests/test_clearmash_api.py::test_get_documents_root_folders INFO:api:_wcm_api_call(/Document/Folder/Root, None)
PASSED
tests/test_clearmash_api.py::test_get_web_document_system_folder_places INFO:api:_wcm_api_call(/Document/Folder/Get, {'FolderId': 43, 'FolderType': 4})
PASSED
tests/test_clearmash_api.py::test_get_documents_places INFO:api:_wcm_api_call(/Documents/Get, {'entitiesIds': [115325, 115353, 115365, 115371, 115388, 265694]})
PASSED
tests/test_clearmash_api.py::test_get_related_docs_of_item INFO:api:_wcm_api_call(/Document/ByRelationField, {'EntityId': 220590, 'FieldId': '_c6_beit_hatfutsot_bh_base_template_multimedia_photos', 'MaxNestingDepth': 1})
FAILED

======================================================= FAILURES =======================================================
____________________________________________ test_get_related_docs_of_item _____________________________________________

    def test_get_related_docs_of_item():
>       related = MockClearmashApi().get_document_related_docs_by_fields(220590, "_c6_beit_hatfutsot_bh_base_template_multimedia_photos")

tests/test_clearmash_api.py:160: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
datapackage_pipelines_mojp/clearmash/api.py:76: in get_document_related_docs_by_fields
    return self._wcm_api_call("/Document/ByRelationField", {"EntityId": entity_id, "FieldId": field, "MaxNestingDepth": max_nesting_depth})
datapackage_pipelines_mojp/clearmash/api.py:82: in _wcm_api_call
    post_data=post_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_clearmash_api.MockClearmashApi object at 0x103a31908>
url = 'https://bh.clearmash.com/API/V5/Services/WebContentManagement.svc/Document/ByRelationField'
headers = {'ClientToken': 'FAKE_TOKEN', 'Content-Type': 'application/json'}
post_data = {'EntityId': 220590, 'FieldId': '_c6_beit_hatfutsot_bh_base_template_multimedia_photos', 'MaxNestingDepth': 1}

    def _get_request_json(self, url, headers, post_data=None):
        if url == "https://bh.clearmash.com/API/V5/Services/WebContentManagement.svc/CommunityPage/Folder/Root":
            filename = os.path.join(os.path.dirname(__file__), "mocks", "clearmash-api-communitypage-folder-root.json")
        elif url == "https://bh.clearmash.com/API/V5/Services/WebContentManagement.svc/invalid/path":
            filename = os.path.join(os.path.dirname(__file__), "mocks", "clearmash-api-invalid-path-response.json")
        elif url == "https://bh.clearmash.com/API/V5/Services/WebContentManagement.svc/Document/Folder/Root":
            filename = os.path.join(os.path.dirname(__file__), "mocks", "clearmash-api-document-folder-root.json")
        elif url == "https://bh.clearmash.com/API/V5/Services/WebContentManagement.svc/Document/Folder/Get":
            filename = os.path.join(os.path.dirname(__file__), "mocks",
                                    "clearmash-api-document-folder-get-{}.json".format(post_data["FolderId"]))
        elif url == "https://bh.clearmash.com/API/V5/Services/WebContentManagement.svc/Documents/Get":
            filename = os.path.join(os.path.dirname(__file__), "mocks",
                                    "clearmash-api-documents-get-{}.json".format("-".join(map(str, post_data["entitiesIds"]))))
        elif url == "https://bh.clearmash.com/API/V5/Services/WebContentManagement.svc/Document/ByRelationField":
            filename = os.path.join(os.path.dirname(__file__), "mocks", "clearmash-api-get-related-docs-by-item-field-{}.json".format(post_data["EntityId"]))
    
        else:
            filename = None
        if filename:
            def get_res():
                try:
                    return super(MockClearmashApi, self)._get_request_json(url, headers, post_data)
                except HTTPError as e:
                    return {"__MOJP_ERROR__": True,
                            "status_code": e.response.status_code,
                            "content": e.response.content.decode()}
            res = self._get_mock_url_request_json(filename, get_res)
            if isinstance(res, dict) and res.pop("__MOJP_ERROR__", False):
>               raise HTTPError(response=type("MockResponse", (object,), res))
E               requests.exceptions.HTTPError

tests/test_clearmash_api.py:55: HTTPError
```

### CODE REVIEW

File: tests/mocks/clearmash-api-get-related-docs-by-item-field-220590.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
+}
```
Comment (OriHoch) [inline tests/mocks/clearmash-api-get-related-docs-by-item-field-220590.json]: @Libisch this is an error response, need to make sure you get the right response from clearmash API first

File: tests/test_clearmash_api.py [hunk 2]
```diff
@@ -39,6 +37,9 @@ def _get_request_json(self, url, headers, post_data=None):
         elif url == "https://bh.clearmash.com/API/V5/Services/WebContentManagement.svc/Documents/Get":
             filename = os.path.join(os.path.dirname(__file__), "mocks",
                                     "clearmash-api-documents-get-{}.json".format("-".join(map(str, post_data["entitiesIds"]))))
+        elif url == "https://bh.clearmash.com/API/V5/Services/WebContentManagement.svc/Document/ByRelationField":
+            filename = os.path.join(os.path.dirname(__file__), "mocks", "clearmash-api-get-related-docs-by-item-field-{}.json".format(post_data["EntityId"]))
```
Comment (OriHoch) [inline tests/test_clearmash_api.py]: need to add the field and the max nesting depth to the filename as well

File: tests/mocks/clearmash-api-get-related-docs-by-item-field-220590.json (hunk 1 above)
Comment (OriHoch) [inline tests/mocks/clearmash-api-get-related-docs-by-item-field-220590.json]: inside that error it says `Client token is invalid`

**Comment by Libisch** [comment]
fixed and replaced by #57

**Pull request closed without merging**